### ISSUE TITLE
Add classes to items that commonly need specific selectors

### ISSUE THREAD

**Issue by jlandfried** [LastCallMedia/Drupal-Scaffold#25]
Right now we're not adding any classes to things like nodes and views. It's helpful to have some basic top level classes for styling purposes. I see these being added through preprocess functions in `scaffold.theme`.

ex:
```
/**
 * Implements hook_preprocess_node().
 */
function scaffold_preprocess_node(&$vars) {
  $vars['attributes']['class'][] = 'node-type-' . $vars['node']->bundle();
}
```
Here's a starting place for things that would be great to have classes on (feel free to add others that'd be useful):

### Nodes

- [x] Bundle: `node-type-{bundle}` 
- [x] Status: `status-{status}`
- [x] View Mode: `view-mode-{view_mode}`

### Views

- [x] View ID: `view-{view_id}`
- [x] Views Display: `view-display-{view_display}`

### Field

- [x] Field Name: `field-name-{field_name}`

**Comment by rbayliss** [comment]
@jlandfried Any thoughts on doing this in templates vs doing it in preprocess?  I think I agree that preprocess is right here, I just want to make sure we document why we're doing it this way, and when people should use preprocess vs adding classes in templates.  I could see this becoming a pattern people repeat, and I'm not sure I agree with it for most use cases.

**Comment by jlandfried** [comment]
@rbayliss yeah, in this case preprocess makes sense in my mind because they're generic classes that we never will really _don't_ want to have. If they don't end up being used then it's not a big deal, but we'll end up repeating things like `{{ attributes.addClass(node.bundle()) }}` a lot if this gets handled at the template level.

My generic thought on this is if it's information that should be present on the majority of the items being rendered, or something that needs to be maintained even through template inheritance (like data-attributes provided by a module), then preprocess is the best place for it. If it's specific to the individual instance that's being rendered/themed then the template's a better place for it.

**Comment by jlandfried** [comment]
@rbayliss actually, looking at how classy does it I wonder if maybe templates _would_ make sense here. They handle classes in the template. Maybe doing that and leveraging twig template extension could give us the best of both worlds?

Something like:
`@scaffold/node.html.twig`:
```
{%
set classes = [
'node',
'node--type-' ~ node.bundle|clean_class,
node.isPromoted() ? 'node--promoted',
node.isSticky() ? 'node--sticky',
not node.isPublished() ? 'node--unpublished',
view_mode ? 'node--view-mode-' ~ view_mode|clean_class,
]
%}

{% block content %}
  <article{{ attributes.addClass(classes) }}>
    {# Node content goes here. #}
  </article>
{% endblock %}
```

And for overrides we could recommend something like:
`@scaffold/node--article.html.twig`:
```
{% extends '@scaffold/content/node.html.twig' %}

{% block content %}
    <div {{ attributes.addClass(classes) }}>
        {# Node content goes here. #}
    </div>
{% endblock %}

```

Then it's easy to not use the classes if you don't want them, just don't extend the base template.

-----

FWIW: This is a thing https://www.drupal.org/node/2387069. So we'd end up maintaining all of the base templates that we wanted this pattern to work for, or we'd have use the theme registry to look up the parent theme's base template if we wanted to use it. Something like this I guess?

```
{%
set classes = [
'node',
'node--type-' ~ node.bundle|clean_class,
node.isPromoted() ? 'node--promoted',
node.isSticky() ? 'node--sticky',
not node.isPublished() ? 'node--unpublished',
view_mode ? 'node--view-mode-' ~ view_mode|clean_class,
]
%}

{% block content %}
  {% set attributes = attributes.addClass(classes) %}
  {% include '@stable/content/node.html.twig' %}
{% endblock %}
```

**Comment by rbayliss** [comment]
I'm definitely on board with this, especially if we're only providing templates for a few key pieces (blocks, nodes, etc).  I think I prefer the first form from your comment above rather than calling back to a base theme... It would be nice to avoid extra trips to the theme registry, which I think are pretty expensive, and really, how often is that markup gonna change?

**Comment by rbayliss** [comment]
Oh - this one got closed a while back by #73!
